components/GifRenderer/common/src/commonMain/kotlin/org/jetbrains/compose/gifrenderer/GifRenderer.kt

```diff
@@ -0,0 +1,12 @@
+package org.jetbrains.compose.gifrenderer
+
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.State
+import androidx.compose.ui.graphics.painter.Painter
+
+@Composable
+fun rememberGifRenderer(url: String): State<Painter> {
```

Comment:
I think we need to provide a way to load gif from any `InputStream` and from any non-composable function (that can be called from `Dispatchers.IO`)

```
fun loadAnimation(inputStream: InputStream): Animation

// load synchronously from resources (see https://android-review.googlesource.com/c/platform/frameworks/support/+/1576433/2/compose/ui/ui/src/desktopMain/kotlin/androidx/compose/ui/res/DesktopSvgResources.desktop.kt#71)
@Composable
fun animationResource(resourcePath: String)

@Composable
fun Animation(animation: Animation) {

}

interface Animation {
  val frames: Int
  fun painterFor(frame: Int): Painter
}
```

P.S. `InputStream` is jvm-only, so it should be part of the `desktopMain`. 
Maybe we can implement something using https://github.com/Kotlin/kotlinx-io, but it is experimental, so our function should be experimental too.